contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol

```diff
@@ -317,4 +328,32 @@ abstract contract UpgradeableTokenPool is IPool, OwnerIsCreator, IERC165 {
     if (IARM(i_armProxy).isCursed()) revert BadARMSignal();
     _;
   }
+
+  /// @notice Getter for proxy pool address.
+  /// @param remoteChainSelector The remote chain selector for which the proxy pool is being retrieved.
+  /// @return proxyPool The proxy pool address for the given remoteChainSelector
+  function getProxyPool(uint64 remoteChainSelector) public view returns (address proxyPool) {
+    assembly ("memory-safe") {
+      mstore(0, PROXY_POOL_SLOT)
+      mstore(32, remoteChainSelector)
+      proxyPool := shr(96, shl(96, sload(keccak256(0, 64))))
+    }
+  }
+
+  /// @notice Setter for proxy pool address, only callable by the DAO.
+  /// @param remoteChainSelector The remote chain selector for which the proxy pool is being set.
+  /// @param proxyPool The address of the proxy pool.
+  function setProxyPool(uint64 remoteChainSelector, address proxyPool) external onlyOwner {
+    _setPoolProxy(remoteChainSelector, proxyPool);
+  }
+
+  function _setPoolProxy(uint64 remoteChainSelector, address proxyPool) internal {
+    if (!isSupportedChain(remoteChainSelector)) revert ChainNotAllowed(remoteChainSelector);
+    if (getProxyPool(remoteChainSelector) != address(0)) revert ProxyPoolAlreadySet(remoteChainSelector);
```

Comment:
I think it's fine to replace the proxy pool value in case it's necessary. This is only callable by `onlyOnwer`, so safe to assume it's intented.

Reply:
sure yep [60e78e8](https://github.com/aave/ccip/pull/18/commits/60e78e8ed5a74b87d3bfe545ac0450b71dab7d6d)